operations/scheduler.go

```diff
@@ -694,6 +697,32 @@ func (s *Scheduler) executeOperationPreconditions(ctx context.Context, operation
 	return nil
 }
 
+func (s *Scheduler) FindConcurrentOperation(ctx context.Context, operation *types.Operation) (*types.Operation, error) {
+	criteria := []query.Criterion{
+		query.ByField(query.EqualsOperator, "resource_id", operation.ResourceID),
+		query.ByField(query.EqualsOperator, "resource_type", string(operation.ResourceType)),
+		query.ByField(query.EqualsOperator, "type", string(operation.Type)),
+		query.OrderResultBy("updated_at", query.DescOrder),
+	}
+	sameResourceOperations, err := s.repository.List(ctx, types.OperationType, criteria...)
+	if err != nil {
+		return nil, err
+	}
+	if sameResourceOperations.Len() > 0 {
+		same := sameResourceOperations.ItemAt(0).(*types.Operation)
+		if same.InOrphanMitigationState() {
+			return same, nil
+		}
+		switch same.State {
```

Comment:
(low prio) why switch? if all options result in the same return value, why not use if with or? also applies for the InOrphanMitigationState check

Reply:
It was refactored to filter using query itself instead having compute logic